src/fix/fixField.h

```diff
@@ -0,0 +1,371 @@
+#pragma once
+#include <string>
+#include <sstream>
+#include <stdint.h>
+#include "cdr.h"
+#include "utils.h"
+#include "codec.h"
+
+#define FIX_MONTHYEAR_FMT "%Y%m"
+#define FIX_UTCTIMEONLY_FMT "%H:%M:%S"
+#define FIX_UTCDATE_FMT "%Y%m%d"
+#define FIX_LOCALMKTDATE_FMT "%Y%m%d"
+#define FIX_UTCTIMESTAMP_FMT "%Y%m%d-%H:%M:%S"
+
+
+namespace neueda
+{
+
+using namespace std;
+
+class fixField;
+
+typedef bool (fixField::*getAsFunc) (const char*, const int64_t, cdr&);
+typedef codecState (fixField::*setAsFunc) (const int64_t,
+                                           const cdrItem&,
+                                           const size_t len,
+                                           char* p,
+                                           size_t& used);
+
+class fixField
+{
+public:
+    fixField () :
+        mRequired (false),
+        mTag (0)
+    {
+        string t ("STRING");
+        setType (t);
+    }
+
+    bool getAsString (const char* val, const int64_t tag, cdr& d)
+    {
+        string value (val);
+        utils_trimWhiteSpace (value);
+
+        d.setString (tag, value);
+
+        return true;
+    }
+
+    static codecState writeStringVal (const int64_t tag,
+                                      const string& val,
+                                      const size_t len,
+                                      char* p,
+                                      size_t& used)
+    {
+        char* tmp = p + used;
+        tmp += snprintf (p + used, len - used, "%lld=%s\001", tag, val.c_str ());
+        used += tmp - (p + used);
+
+        if (used > len)
+            return GW_CODEC_SHORT;
+
+        return GW_CODEC_SUCCESS;
+    }
+
+    static codecState setAsString (const int64_t tag,
+                                   const cdrItem& i,
+                                   const size_t len,
+                                   char* p,
+                                   size_t& used)
+    {
+        string val;
+        if (!i.asString(val))
+            return GW_CODEC_ERROR;
+
+        return writeStringVal (tag, val, len, p, used);
+    }
+
+    bool get (const char* val, cdr& d)
+    {
+        return ((this)->*mGetter) (val, mTag, d);
+    }
+
+    codecState set (const int64_t tag,
+                    const cdrItem& i,
+                    const size_t len,
+                    char* p,
+                    size_t& used)
+    {
+        if (mSetter != NULL)
+            return ((this)->*mSetter) (tag, i, len, p, used);
+        else
+            return fixField::setAsString (tag, i, len, p, used);
+    }
+
+    bool required () const
+    {
+        return mRequired;
+    }
+
+    void setRequired (bool req)
+    {
+        mRequired = req;
+    }
+
+    int64_t tag () const
+    {
+        return mTag;
+    }
+
+    void setTag (int64_t tag)
+    {
+        mTag = tag;
+    }
+
+    string getName () const
```

Comment:
namining here is a bit inconsistent. Tag getter is `tag()`, but the name getter is `getName()`, also please return const std::string&